minlzlib/xzstream.c

```diff
@@ -31,6 +31,19 @@ Module Name:
 #include "minlzlib.h"
 #include "xzstream.h"
 
+//
+// XzDecodeBlockHeader can return "I successfully found a block",
+// "I failed/bad block header", or "there was no block header".
+// Though minlzlib explicitly only claims to handle files with a
+// single block, it needs to also handle files with no blocks at all.
+// (Produced by "xz" when compressing an empty input file)
+//
+typedef enum _XZ_DECODE_BLOCK_HEADER_RESULT {
```

Comment:
I have made an explicit decision not to include defined error codes to keep code smaller. I understand this comes at a debuggability cost, but undoing that decision should be

1) Outside of the scope of this PR
2) If decided upon, done for more than just this particular scenario, because many other classes of errors exist.

I'd appreciate eliding the set of changes that pertain to this particular scope. Thank you!